### PR TITLE
Add `go vet` to Travis CI

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,9 +9,11 @@ go:
 before_install:
   - go get -u github.com/golang/lint/golint
   - if [[ $TRAVIS_GO_VERSION == 1.5* ]]; then go get -u github.com/kisielk/errcheck; fi
+  - go get -u golang.org/x/tools/cmd/vet
```
Comment (shurcooL) [inline .travis.yml]: Is this necessary and desired? Go 1.6 ships with `vet` built-in.

Comment (tamird) [inline .travis.yml]: Yes. The alternative is only to run `go vet` in 1.6, but minimizing dependence on `$TRAVIS_GO_VERSION` seems like a win.

**Comment by soheilhy** [comment]
I agree that it's worth "vet"'ing the code with other versions of go.

**Comment by soheilhy** [comment]
LGTM. Thanks @tamird!

**Pull request merged**